### PR TITLE
Fixup samples

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

File: samples/guide/src/test/kotlin/okhttp3/AllMainsTest.kt [hunk 1]
```diff
@@ -23,14 +23,18 @@ import java.io.File
 import java.lang.reflect.InvocationTargetException
 
 @RunWith(Parameterized::class)
-@Ignore
+//@Ignore
```
Comment (yschimke) [inline samples/guide/src/test/kotlin/okhttp3/AllMainsTest.kt]: TODO: revert before landing

File: samples/guide/src/main/java/okhttp3/recipes/kt/CustomTrust.kt [hunk 2]
```diff
@@ -23,24 +23,19 @@ import java.io.IOException
 import java.io.InputStream
 import java.security.KeyStore
 import java.security.cert.CertificateFactory
-import java.util.Arrays
-import javax.net.ssl.KeyManagerFactory
-import javax.net.ssl.SSLContext
-import javax.net.ssl.TrustManager
-import javax.net.ssl.TrustManagerFactory
-import javax.net.ssl.X509TrustManager
+import javax.net.ssl.*
```
Comment (yschimke) [inline samples/guide/src/main/java/okhttp3/recipes/kt/CustomTrust.kt]: TODO: revert

File: samples/guide/src/main/java/okhttp3/recipes/PostStreamingWithPipe.java [hunk 3]
```diff
@@ -32,6 +32,8 @@
   private final OkHttpClient client = new OkHttpClient();
 
   public void run() throws Exception {
+    if (true) return;
```
Comment (yschimke) [inline samples/guide/src/main/java/okhttp3/recipes/PostStreamingWithPipe.java]: TODO: revert

File: samples/guide/src/main/java/okhttp3/recipes/CustomTrust.java [hunk 4]
```diff
@@ -39,21 +41,30 @@
...
+
   public CustomTrust() {
-    X509TrustManager trustManager;
     SSLSocketFactory sslSocketFactory;
     try {
-      trustManager = trustManagerForCertificates(trustedCertificatesInputStream());
+      KeyAndTrustManagers keyAndTrustManagers = trustManagerForCertificates(trustedCertificatesInputStream());
       SSLContext sslContext = SSLContext.getInstance("TLS");
-      sslContext.init(null, new TrustManager[] { trustManager }, null);
+      sslContext.init(keyAndTrustManagers.keyManagers, keyAndTrustManagers.trustManagers, null);
```
Comment (yschimke) [inline samples/guide/src/main/java/okhttp3/recipes/CustomTrust.java]: @swankjesse Should we change this test to use okhttp-tls?

**Comment by yschimke** [comment]
@swankjesse Did you change your certificates for publicobject.com to Let's Encrypt?

image

In general, how can we make it easier to debug a failure like this?

**Comment by yschimke** [comment]
Root certs from the test

image


image

File: samples/guide/src/main/java/okhttp3/recipes/CustomTrust.java (hunk 4 above)
Comment (swankjesse) [reply samples/guide/src/main/java/okhttp3/recipes/CustomTrust.java]: I did!

Comment (swankjesse) [reply samples/guide/src/main/java/okhttp3/recipes/CustomTrust.java]: Yes!

**Comment by swankjesse** [comment]
Yep, I changed publicobject to letsencrypt. Good catch.

**Comment by yschimke** [comment]
@swankjesse it really shouldn't require this much expert knowledge to diagnose, I think we can do a lot better to explain these problems.  Just not sure how...

File: samples/guide/src/main/java/okhttp3/recipes/CustomTrust.java (hunk 4 above)
Comment (yschimke) [reply samples/guide/src/main/java/okhttp3/recipes/CustomTrust.java]: For followup https://github.com/square/okhttp/issues/5701

**Pull request merged**